### PR TITLE
Add ValidatingWebhookConfiguration for PA-614

### PR DESCRIPTION
**Pull request by tthurlapati-px**
**What this PR does / why we need it**:
Adds `ValidatingWebhookConfigurationOps` and `ValidatingWebhookConfigurationV1beta1Ops` to `admissionRegistration` in order to provide support to functions in Torpedo as part of PA-614.

It also refactors (renames) some of the imports in `addmissionRegistration` to have consistent naming for `v1` and `v1beta1`.

**Special notes for your reviewer**:
It has been tested as part of PA-614 (in Torpedo). Refer https://github.com/portworx/torpedo/pull/1182 (optional)

### CODE REVIEW

File: k8s/admissionregistration/validatingwebhooks_v1beta1.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+)
+
+// ValidatingWebhookConfigurationV1beta1Ops is interface to perform CRUD ops on mutatting webhook controller
+type ValidatingWebhookConfigurationV1beta1Ops interface {
+	// GetValidatingWebhookConfigurationV1beta1 returns a given ValidatingWebhookConfiguration
+	GetValidatingWebhookConfigurationV1beta1(name string) (*hook.ValidatingWebhookConfiguration, error)
+	// CreateValidatingWebhookConfigurationV1beta1 creates given ValidatingWebhookConfiguration
+	CreateValidatingWebhookConfigurationV1beta1(req *hook.ValidatingWebhookConfiguration) (*hook.ValidatingWebhookConfiguration, error)
+	// UpdateValidatingWebhookConfigurationV1beta1 updates given ValidatingWebhookConfiguration
+	UpdateValidatingWebhookConfigurationV1beta1(*hook.ValidatingWebhookConfiguration) (*hook.ValidatingWebhookConfiguration, error)
```
Comment (nikolaypopov) [inline k8s/admissionregistration/validatingwebhooks_v1beta1.go:17]: Can you either add req to both functions or remove from both function please, let's be consistent

File: k8s/admissionregistration/validatingwebhooks_v1beta1.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	// DeleteValidatingWebhookConfigurationV1beta1 deletes given ValidatingWebhookConfiguration
+	DeleteValidatingWebhookConfigurationV1beta1(name string) error
+}
+
+// GetValidatingWebhookConfigurationV1beta1 returns a given ValidatingWebhookConfiguration
```
Comment (nikolaypopov) [inline k8s/admissionregistration/validatingwebhooks_v1beta1.go:22]: nit: Remove `a` to be consistent

File: k8s/admissionregistration/validatingwebhooks_v1beta1.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import (
+	"context"
+
+	hook "k8s.io/api/admissionregistration/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ValidatingWebhookConfigurationV1beta1Ops is interface to perform CRUD ops on mutatting webhook controller
+type ValidatingWebhookConfigurationV1beta1Ops interface {
+	// GetValidatingWebhookConfigurationV1beta1 returns a given ValidatingWebhookConfiguration
```
Comment (nikolaypopov) [inline k8s/admissionregistration/validatingwebhooks_v1beta1.go:12]: nit: Remove `a` to be consistent

File: k8s/admissionregistration/validatingwebhooks.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import (
+	"context"
+
+	hook "k8s.io/api/admissionregistration/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ValidatingWebhookConfigurationOps is interface to perform CRUD ops on mutatting webhook controller
+type ValidatingWebhookConfigurationOps interface {
+	// GetValidatingWebhookConfiguration returns a given ValidatingWebhookConfiguration
```
Comment (nikolaypopov) [inline k8s/admissionregistration/validatingwebhooks.go:12]: nit: Remove `a` to be consistent

**Review by nikolaypopov — COMMENTED** [review]
LGTM, please address minor comments and rebase

File: k8s/admissionregistration/validatingwebhooks_v1beta1.go (hunk 1 above)
Comment (tthurlapati-px) [reply k8s/admissionregistration/validatingwebhooks_v1beta1.go:17]: ✅

File: k8s/admissionregistration/validatingwebhooks_v1beta1.go (hunk 2 above)
Comment (tthurlapati-px) [reply k8s/admissionregistration/validatingwebhooks_v1beta1.go:22]: ✅

File: k8s/admissionregistration/validatingwebhooks_v1beta1.go (hunk 3 above)
Comment (tthurlapati-px) [reply k8s/admissionregistration/validatingwebhooks_v1beta1.go:12]: ✅

File: k8s/admissionregistration/validatingwebhooks.go (hunk 4 above)
Comment (tthurlapati-px) [reply k8s/admissionregistration/validatingwebhooks.go:12]: ✅

File: k8s/admissionregistration/validatingwebhooks.go [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+	GetValidatingWebhookConfiguration(name string) (*hook.ValidatingWebhookConfiguration, error)
+	// CreateValidatingWebhookConfiguration creates given ValidatingWebhookConfiguration
+	CreateValidatingWebhookConfiguration(req *hook.ValidatingWebhookConfiguration) (*hook.ValidatingWebhookConfiguration, error)
+	// UpdateValidatingWebhookConfiguration updates given ValidatingWebhookConfiguration
+	UpdateValidatingWebhookConfiguration(*hook.ValidatingWebhookConfiguration) (*hook.ValidatingWebhookConfiguration, error)
+	// DeleteValidatingWebhookConfiguration deletes given ValidatingWebhookConfiguration
+	DeleteValidatingWebhookConfiguration(name string) error
+}
+
+// GetValidatingWebhookConfiguration returns a given ValidatingWebhookConfiguration
```
[the remark this answers is not in the document]
Comment (tthurlapati-px) [reply k8s/admissionregistration/validatingwebhooks.go:22]: ✅

**Comment by tthurlapati-px** [comment]
> LGTM, please address minor comments and rebase

changes made

**Review by nikolaypopov — APPROVED** [review]
LGTM

**Pull request merged**